src/v/utils/external_process.h

```diff
@@ -0,0 +1,262 @@
+    process_already_being_terminated,
+    failed_to_terminate_process,
+    terminate_called_before_wait
+};
+
+/**
+ * @brief Defines the external_proces::error_category
+ */
+struct error_category final : public std::error_category {
+    /**
+     * @return const char* Name of the category
+     */
+    const char* name() const noexcept final {
+        return "external_process::error_category";
+    }
+
+    /**
+     * @param c Error code to stringify
+     * @return std::string Stringified version of @p c
+     */
+    std::string message(int c) const final {
+        switch (static_cast<error_code>(c)) {
+        case error_code::success:
+            return "success";
+        case error_code::process_already_completed:
+            return "process already completed";
+        case error_code::process_already_being_waited_on:
+            return "wait already called on process";
+        case error_code::process_already_being_terminated:
+            return "terminate already being called on the process";
+        case error_code::failed_to_terminate_process:
+            return "failed to terminate the process";
+        case error_code::terminate_called_before_wait:
+            return "termiate called before wait";
+        }
+
+        return "(unknown error code)";
+    }
+};
+
+/**
+ * @return const std::error_category& Returns static instance of
+ * external_process::error_category
+ */
+inline const std::error_category& error_category() noexcept {
+    static struct error_category e;
+    return e;
+}
+inline std::error_code make_error_code(error_code e) noexcept {
+    return {static_cast<int>(e), error_category()};
+}
+
+/**
+ * @brief This class wraps a Seastar experimental process instance
+ *
+ * The purpose of this class is to provide an easy way of controlling a spawned
+ * child process.
+ */
+class external_process final {
+public:
+    /**
+     * @brief Create a external process object
+     *
+     * This method will return a created external_process object with an already
+     * spawned process.  It will attach any provided consumer to the appropriate
+     * ss::input_stream<char>.
+     *
+     * @note You _must_ call `wait()` on the returned object before the object
+     * is destructed.
+     *
+     * @param command The command to run with its arguments, divided over the
+     * vector.  Will assert if empty.
+     * @param env Optionall provided a list of environmental variables as
+     * KEY=VALUE
+     * @return A created external process
+     * @throw std::system_error If unable to launch the process
+     */
+    static ss::future<external_process> create_external_process(
+      std::vector<ss::sstring> command,
+      std::optional<std::vector<ss::sstring>> env = std::nullopt);
+
+    external_process() = delete;
+    external_process(const external_process&) = delete;
+    external_process& operator=(const external_process&) = delete;
+    external_process(external_process&&) noexcept;
```

Comment:
Why support move constructing only?

Reply:
Oh this was back when I originally had the move constructor set to `default`.  `ss::experimental::process`'s move operator is implicitly deleted.  I can probably write one.

Reply:
Attempting to write move assignment runs into issues with `ss::experimental::process`:

```c++
external_process& external_process::operator=(external_process&& rhs) noexcept {
    _process = std::move(rhs._process); // <--- cannot be assigned because its copy assignment operator is implicitly deleted
}
```

Reply:
Should we just force pointer stability and return this object in a unique_ptr then?

Reply:
Let's just use `unique_ptr` here then.

Reply:
You mean make `_process` a `std::unique_ptr<ss::experimental::process>`?

Reply:
Or make the static factory function return a `std::unique_ptr<external_process>` (my preference).